clients/filesystem-fuse/src/s3_filesystem.rs

```diff
@@ -0,0 +1,268 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+use crate::config::AppConfig;
+use crate::error::ErrorCode::{InvalidConfig, OpenDalError};
+use crate::filesystem::{FileStat, FileSystemCapacity, FileSystemContext, PathFileSystem, Result};
+use crate::gravitino_client::{Catalog, Fileset};
+use crate::open_dal_filesystem::OpenDalFileSystem;
+use crate::opened_file::{OpenFileFlags, OpenedFile};
+use crate::utils::{parse_location, GvfsResult};
+use async_trait::async_trait;
+use log::error;
+use opendal::layers::LoggingLayer;
+use opendal::services::S3;
+use opendal::{Builder, Operator};
+use std::collections::HashMap;
+use std::path::Path;
+
+pub(crate) struct S3FileSystem {
+    open_dal_fs: OpenDalFileSystem,
+}
+
+impl S3FileSystem {}
+
+impl S3FileSystem {
+    const S3_CONFIG_PREFIX: &'static str = "s3-";
+
+    pub(crate) fn new(
+        catalog: &Catalog,
+        fileset: &Fileset,
+        config: &AppConfig,
+        _fs_context: &FileSystemContext,
+    ) -> GvfsResult<Self> {
+        let mut opendal_config = extract_s3_config(config);
+        let bucket = extract_bucket(&fileset.storage_location)?;
+        opendal_config.insert("bucket".to_string(), bucket);
+
+        let endpoint = catalog.properties.get("s3-endpoint");
```

Comment:
`s3-endpoint` may be empty for fileset,  cc @yuqi1129

Reply:
please try get `s3-region` first,  maybe we could use a new func like `get_s3_region()`